queries/sun_position.py

```diff
@@ -0,0 +1,623 @@
+    """
+    Parse text received from Almanak HÍ endpoint into usable dict.
+    """
+    data: _SOLAR_DICT_TYPE = {}
+    city: Optional[str] = None
+    month: Optional[int] = None
+
+    for line in text:
+        city_re = re.match(_ALMANAK_CITY_REGEX, line)
+        if city_re:
+            # Matched line containing name of city
+            city, lat, lon = city_re.groups()
+            city = capitalize_placename(city.lower())
+
+            # Initialize dict for Icelandic city (place)
+            data[city] = {"pos": _convert_dms_lat_lon_to_decimal(lat, lon)}
+
+        else:
+            sun_re = re.match(_ALMANAK_SUNPOS_REGEX, line)
+            if sun_re and city is not None:
+                # Matched line containing times of solar positions
+                if sun_re.group(1) != " - ":
+                    # New month started
+                    month_str = sun_re.group(1).lower()
+                    month = MONTH_ABBREV.index(month_str) + 1
+
+                # Extract times of solar positions
+                sun_pos: _SOLAR_ROW_TYPE = dict(
+                    zip(
+                        _ALMANAK_HI_COLUMNS,
+                        (_parse_almanak_cell(cell) for cell in sun_re.groups()[2:]),
+                    )
+                )
+
+                # Calculate solar midnight from solar noon
+                # ("Hádegi" (solar noon) is never None)
+                solar_noon: datetime.time = cast(
+                    datetime.time, sun_pos[_SOLAR_POSITIONS.HÁDEGI]
+                )
+                solar_midnight = datetime.time(
+                    hour=((solar_noon.hour + 12) % 24), minute=solar_noon.minute
+                )
+                sun_pos[_SOLAR_POSITIONS.MIÐNÆTTI] = solar_midnight
+
+                # Add solar positions for city on a specific date
+                date = datetime.date.today().replace(
+                    month=cast(int, month), day=int(sun_re.group(2))
+                )
+                data[city][date] = sun_pos
+
+    return data
+
+
+@cached(TTLCache(maxsize=1, ttl=86400))
+def _get_almanak_hi_data() -> _SOLAR_DICT_TYPE:
+    r = requests.get(_ALMANAK_HI_URL)
+    # Use beautiful soup to extract text from HTML response
+    # and split on newlines
+    text: List[str] = (
+        BeautifulSoup(r.text, "html.parser")
+        .get_text()
+        .replace("\r\n", "\n")
+        .split("\n")
+    )
+    return _parse_almanak_hi_data(text)
+
+
+def _find_closest_city(data: _SOLAR_DICT_TYPE, loc: LatLonTuple) -> Optional[str]:
+    """Find city closest to loc in data."""
+    closest_city = None
+    closest_distance = None
+
+    for city_name, city_dict in data.items():
+        dist = distance(loc, cast(LatLonTuple, city_dict["pos"]))
+
+        if closest_distance is None or dist < closest_distance:
+            closest_distance = dist
+            closest_city = city_name
+
+    return closest_city
+
+
+_SOLAR_ENUM_TO_WORD: Dict[_SOLAR_POS_ENUM, str] = {
```

Comment:
Move these constants to the beginning of the file?

Reply:
Moved the constants together for more clarity.